src/rewrite.rs

```diff
@@ -30,6 +31,23 @@ impl<T: Rewrite> Rewrite for ptr::P<T> {
     }
 }
 
+#[derive(Clone, Debug)]
+pub(crate) enum MacroErrorKind {
+    ParseFailure,
+    ReplaceMacroVariable,
+    Unknown,
+}
+
+impl fmt::Display for MacroErrorKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            MacroErrorKind::ParseFailure => write!(f, "(parse failure)"),
+            MacroErrorKind::ReplaceMacroVariable => write!(f, "(replacing macro variables with $)"),
+            MacroErrorKind::Unknown => write!(f, ""),
+        }
+    }
+}
+
```

Comment:
Including `Unknown` variant again in MacroErrorKind might be undesirable, but at least we have `span` field for unknown macro error.

Reply:
I'm fine with an `Unknown` variant as a catch all for any other kind of unexpected error. We can always add more variants in the future.